src/ripple/core/JobTypes.h

```diff
@@ -21,171 +21,147 @@
-        add(jtWAL,               "writeAhead",           maxLimit,  1000ms,  2500ms);
-        add(jtVALIDATION_t,      "trustedValidation",    maxLimit,   500ms,  1500ms);
-        add(jtWRITE,             "writeObjects",         maxLimit,  1750ms,  2500ms);
-        add(jtACCEPT,            "acceptLedger",         maxLimit,     0ms,     0ms);
-        add(jtPROPOSAL_t,        "trustedProposal",      maxLimit,   100ms,   500ms);
-        add(jtSWEEP,             "sweep",                       1,     0ms,     0ms);
-        add(jtNETOP_CLUSTER,     "clusterReport",               1,  9999ms,  9999ms);
-        add(jtNETOP_TIMER,       "heartbeat",                   1,   999ms,   999ms);
-        add(jtADMIN,             "administration",       maxLimit,     0ms,     0ms);
-        add(jtMISSING_TXN,       "handleHaveTransactions",   1200,     0ms,     0ms);
-        add(jtREQUESTED_TXN,     "doTransactions",           1200,     0ms,     0ms);
-
-        add(jtPEER,              "peerCommand",                 0,   200ms,  2500ms);
-        add(jtDISK,              "diskAccess",                  0,   500ms,  1000ms);
-        add(jtTXN_PROC,          "processTransaction",          0,     0ms,     0ms);
-        add(jtOB_SETUP,          "orderBookSetup",              0,     0ms,     0ms);
-        add(jtPATH_FIND,         "pathFind",                    0,     0ms,     0ms);
-        add(jtHO_READ,           "nodeRead",                    0,     0ms,     0ms);
-        add(jtHO_WRITE,          "nodeWrite",                   0,     0ms,     0ms);
-        add(jtGENERIC,           "generic",                     0,     0ms,     0ms);
-        add(jtNS_SYNC_READ,      "SyncReadNode",                0,     0ms,     0ms);
-        add(jtNS_ASYNC_READ,     "AsyncReadNode",               0,     0ms,     0ms);
-        add(jtNS_WRITE,          "WriteNode",                   0,     0ms,     0ms);
-        // clang-format on
-    }
+/** Defines the fixed attributes of a particular type of job.
 
+    These attributes are defined at compile time and never change. They
+    include things like the name of the job, the limit of concurrent jobs
+    of this type, etc.
+ */
+struct JobTypeInfo
+{
 public:
-    static JobTypes const&
-    instance()
-    {
-        static JobTypes const types;
-        return types;
-    }
-
-    static std::string const&
-    name(JobType jt)
-    {
-        return instance().get(jt).name();
-    }
-
-    JobTypeInfo const&
-    get(JobType jt) const
-    {
-        Map::const_iterator const iter(m_map.find(jt));
-        assert(iter != m_map.end());
-
-        if (iter != m_map.end())
-            return iter->second;
-
-        return m_unknown;
-    }
-
-    JobTypeInfo const&
-    getInvalid() const
-    {
-        return m_unknown;
-    }
-
-    Map::size_type
-    size() const
+    JobType const type;
+
+    /// The job's display name (when we get C++20 this could be std::string)
+    char const* const name;
```

Comment:
I think I'm OK with that, although `std::string_view` is at least 8 bytes larger than a naked pointer. I wonder if there's enough "padding" in the layout to absorb this.

Not that I care about 8 bytes in _this_ case, since even with a 1,000,000 jobs in the queue (an insanely and unfathomably large number of 'em) we'd only be "wasting" 8MB.